fvm/evm/types/events.go

```diff
@@ -21,8 +21,46 @@ const (
 )
 
 type EventPayload interface {
-	// CadenceEvent creates a Cadence event type
-	CadenceEvent() (cadence.Event, error)
+	// ToCadence converts the event to Cadence event
+	ToCadence() (cadence.Event, error)
+}
+
+type BlockEventPayload struct {
+	Height            uint64           `cadence:"height"`
+	Hash              string           `cadence:"hash"`
+	Timestamp         uint64           `cadence:"timestamp"`
+	TotalSupply       cadence.Int      `cadence:"totalSupply"`
+	TotalGasUsed      uint64           `cadence:"totalGasUsed"`
+	ParentBlockHash   string           `cadence:"parentHash"`
+	ReceiptRoot       string           `cadence:"receiptRoot"`
+	TransactionHashes []cadence.String `cadence:"transactionHashes"`
+}
+
+// DecodeBlockEventPayload decodes Cadence event into block event payload.
+func DecodeBlockEventPayload(event cadence.Event) (*BlockEventPayload, error) {
+	var block BlockEventPayload
+	err := cadence.DecodeFields(event, &block)
+	return &block, err
+}
+
+type TransactionEventPayload struct {
+	Hash            string `cadence:"hash"`
+	Index           uint16 `cadence:"index"`
+	TransactionType uint8  `cadence:"type"`
+	Payload         string `cadence:"payload"`
+	Error           uint16 `cadence:"error"`
```

Comment:
Minor: The error is likely to confuse users. I think I should port this [method](https://github.com/onflow/flow-evm-gateway/blob/main/services/requester/requester.go#L494-L564), from the EVM Gateway to `flow-go`, so that other developers can benefit from it. But that's something for a different PR.